### DISCUSSION TITLE
Revert commit as syncing my power app lost everything

### DISCUSSION THREAD

**Discussion by wongshimun** [community/community#110159] [New to GitHub]
i was working on my Microsoft power app with my colleagues, then i wanted to sync to see the changes they made on another screen. 
Suddenly my app turned blank and instead of syncing my changes, it synced the blank app. now everything in gone as the github synced an empty app. was looking into reverting back to the previous states, before the blank app was synced, tried every method, cant seems to work.

i really need help with getting the app back as i am having board meeting soon. any helpers?

**Comment by ThisIsMrIsmail** [comment]
Hello @wongshimun,

That sounds super frustrating. Firstly, tech issues are stressful. Take a deep breath and let's try the following:

**GitHub History:**
Try navigating through GitHub history, it should track past versions of your app. Check for a "revert" or "history" option in the interface. Can you roll back to before the blank version?

**Git Versions:**
- Check if you used git, were you creating commits regularly?
- Look for the git log history: If yes, find the last version before the app went blank.
- Roll back to the version you want using `git revert` command to help you "rewind" to that good version. this website could be helpful: https://sentry.io/answers/revert-a-git-repository-to-a-previous-commit/

Didn't use Git? go to the next step.

**Local Backups:**
Check if you saved a local version on your computer. Search for any ".msapp" files or backups you might have made.

**Reach Out to your Colleagues:**
Check if anyone of your colleagues saved a recent copy of the app. Maybe they have a version you can use as a starting point.

**No Luck?**
Explain to the board, be honest about the tech glitch, and focus on what you can show for the meeting.

Hang in there! You might be able to solve this!
